File: Graphs/Graph.java

package Graphs;
import java.util.ArrayList;
public class Graph{

// list of nodes in the graph   
ArrayList<Node> nodes = new ArrayList<Node>();
public int matrix [][];

public Graph(){

}


// creates graph using matrix approach.
 
    public void setgraph(int vertices,int edges){
        matrix = new int[vertices][edges];

    }



    public void createnode(String name){
        Node node = new Node();
        node.SetName(name);
        node.setconnected(false);
        node.setconnectedto(null);
        nodes.add(node);

    }

    public void addconnection(int _vertices,int _edges){
        // 1 means there is a connection between edges, 0 means there isnt a connection.
matrix[_vertices][_edges] = 1;
for(int i=0;i<nodes.size();i++){
    if(nodes.get(i).getName().equals(_vertices) && nodes.get(i).getName().equals(_edges)){
        nodes.get(i).setconnected(true);
        nodes.get(i).setconnectedto(nodes.get(_edges));
    }
}
System.out.println("Proceso conexion finalizado.");



    }

    public boolean checkconnection(int vertices, int edges){
        boolean connection = false;
       
                if(matrix[vertices][edges] == 1 && matrix[vertices][edges] == 1){
                    connection = true;

                }
                else connection = false;

         
        return connection;

    }

   

    public void addnodes(){
        Graph graph = new Graph();

        // create node with info as parameter.
        graph.createnode("Jose");
        graph.addconnection(0,0);


    }
}   

File: Graphs/Graph.py
import numpy as np
import Node 
class Graph:
    def __init__(self,vertices,edges):
        self.vertices = vertices
        self.edges = edges
        self.matrix = 0

    def creategraph(self):
    
       self.matrix = np.zeros((self.vertices,self.vertices))
       print(self.matrix)


    def connect(self,vertix1,vertix2):
        self.matrix[vertix1][vertix2] = 1
        self.matrix[vertix2][vertix1] = 1
        Node.Node.setconnectedto(vertix2)
        print(self.matrix)


    def disconnect(self,vertix1,vertix2):
        self.matrix[vertix1][vertix2] = 0
        self.matrix[vertix2][vertix1] = 0
        print(self.matrix)


def run():
    people = []
    while True:
        name = input('enter your name')
        age = int(input("enter your age"))
        people.append(1)
        n1 = Node.Node(name,age)
        if name == '' and age == 0:
            break


    

    g1 = Graph(len(people),len(people))
    g1.creategraph()
    g1.connect(0,1)
         


if __name__ == "__main__":
    run()




        

        